installer/pkg/cmd/service_catalog.go

```diff
@@ -450,15 +451,46 @@ func uninstallServiceCatalog(ns string) error {
 
 	defer os.RemoveAll(dir)
 
+	// It might take a while to delete the configs, so we want
+	fmt.Println("deleting service catalog configs...")
 	err = deleteConfig(dir)
 	if err != nil {
 		return fmt.Errorf("error deploying YAML files: %v", err)
 	}
 
+	// Namespaces are deleted asynchronuously and we need to make sure the
+	// deletion is actually done before printing the success message.
+	waitOnNSDeletion()
+
 	fmt.Println("uninstalled service catalog successfully")
 	return nil
 }
 
+// waitOnNSDeletion keeps checking whether namespace "service-catalog" is deleted.
+func waitOnNSDeletion() {
+	baseDelay := 100 * time.Millisecond
+	maxDelay := 6 * time.Second
+	retries := 0
+	var err error
+
+	for err == nil {
```

Comment:
the condition probably not needed if we return straight out on err != nil below

Reply:
Done.